src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,2005 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Symbols.Retargeting;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.CSharp.UnitTests;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.Test;
+
+public class OverloadResolutionPriorityTests : CSharpTestBase
+{
+    [Theory, CombinatorialData]
+    public void IncreasedPriorityWins_01(bool useMetadataReference, bool i1First)
+    {
+        var executable = """
+            I3 i3 = null;
+            C.M(i3);
+            """;
+
+        var i1Source = """
+                [OverloadResolutionPriority(1)]
+                public static void M(I1 x) => System.Console.WriteLine(1);
+            """;
+
+        var i2Source = """
+                public static void M(I2 x) => throw null;
+            """;
+
+        var source = $$"""
+            using System.Runtime.CompilerServices;
+
+            public interface I1 {}
+            public interface I2 {}
+            public interface I3 : I1, I2 {}
+
+            public class C
+            {
+                {{(i1First ? i1Source : i2Source)}}
+                {{(i1First ? i2Source : i1Source)}}
+            }
+            """;
+
+        CompileAndVerify([executable, source, OverloadResolutionPriorityAttributeDefinition], expectedOutput: "1", symbolValidator: module =>
```

Comment:
Consider verifying `OverloadResolutionPriority` in source too (with `sourceSymbolValidator`). Applies to other tests too

Reply:
Gentle ping

Reply:
What was the resolution?